src/meta/processors/indexMan/RebuildEdgeIndexProcessor.cpp

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/ActiveHostsMan.h"
+#include "meta/processors/admin/AdminClient.h"
+#include "meta/processors/indexMan/RebuildEdgeIndexProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void RebuildEdgeIndexProcessor::process(const cpp2::RebuildIndexReq& req) {
+    auto space = req.get_space_id();
+    const auto &indexName = req.get_index_name();
+    auto edgeType = req.get_schema_id().get_edge_type();
+    auto version = req.get_version();
+
+    LOG(INFO) << "Rebuild Edge Index Space " << space << ", Index Name " << indexName;
+
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::edgeIndexLock());
+    std::unique_ptr<AdminClient> client(new AdminClient(kvstore_));
+    auto spaceKey = MetaServiceUtils::spaceKey(space);
+    auto spaceRet = doGet(spaceKey);
+    if (!spaceRet.ok()) {
+        LOG(ERROR) << "Space " << space << " Not Found";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+
+    auto partsRet = client->getPartsDist(space).get();
+    if (!partsRet.ok()) {
+        LOG(ERROR) << "Get space " << space << "'s part failed";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+
+    auto parts = partsRet.value();
+    auto edgeIndexIDResult = getIndexID(space, indexName);
+    if (!edgeIndexIDResult.ok()) {
+        LOG(ERROR) << "Edge Index " << indexName << " Not Found";
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+
+    auto edgeIndexID = edgeIndexIDResult.value();
+    auto edgeKey = MetaServiceUtils::indexKey(space, edgeIndexIDResult.value());
+    auto edgeResult = doGet(edgeKey);
+    if (!edgeResult.ok()) {
+        resp_.set_code(cpp2::ErrorCode::E_NOT_FOUND);
+        onFinished();
+        return;
+    }
+
+    auto blockingStatus = client->blockingWrites(space, SignType::BLOCK_ON).get();
```

Comment:
Do we need to blocking meta server?

Reply:
maybe when building the index, we should block the write or else we can't find which one is unindexed

Reply:
Do we need blocking the meta server ?  let avoid tag drop or tag alter? 
for example : 
`kvstore_->setWriteBlocking(space, true);`